_posts/2021-05-26-#22006.md

```diff
@@ -0,0 +1,95 @@
+- The tracing scripts compile and then load eBPF bytecode into this VM. When attached to a tracepoint, the eBPF program is called with the arguments passed to the tracepoint.
+- Based on your use case, the eBPF program can, for example, filter the data or pass it along to the tracing logic in the tracing script. The eBFP VM is quite limited. For example, it has a stack size of 512 bytes.
+
+[eBPF]: https://ebpf.io
+
+
+```
+                ┌──────────────────┐            ┌──────────────┐
+                │ tracing script   │            │ bitcoind     │
+                │==================│      2.    │==============│
+                │  eBPF  │ tracing │      hooks │              │
+                │  code  │ logic   │      into┌─┤►tracepoint 1─┼───┐ 3.
+                └────┬───┴──▲──────┘          ├─┤►tracepoint 2 │   │ pass args
+            1.       │      │ 4.              │ │ ...          │   │ to eBPF
+    User    compiles │      │ pass data to    │ └──────────────┘   │ program
+    Space    & loads │      │ tracing script  │                    │
+    ─────────────────┼──────┼─────────────────┼────────────────────┼───
+    Kernel           │      │                 │                    │
+    Space       ┌──┬─▼──────┴─────────────────┴────────────┐       │
+                │  │  eBPF program                         │◄──────┘
+                │  └───────────────────────────────────────┤
+                │ eBPF kernel Virtual Machine (sandboxed)  │
+                └──────────────────────────────────────────┘
+
+1. The tracing script complies the eBPF code and loads the eBFP program into a kernel VM
+2. The eBPF program hooks into one or more tracepoints
+3. When the tracepoint is called, the arguments are passed to the eBPF program
+4. The eBPF program processes the arguments and returns data to the tracing script
+```
+
+- More information on eBPF:
+- [collabora.com: An eBPF overview, part 1: Introduction](https://www.collabora.com/news-and-blog/blog/2019/04/05/an-ebpf-overview-part-1-introduction/)
+- [collabora.com: An eBPF overview, part 2: Machine & bytecode](https://www.collabora.com/news-and-blog/blog/2019/04/15/an-ebpf-overview-part-2-machine-and-bytecode/)
+- [Brendan D. Gregg's blog posts, and book on on eBPF](http://www.brendangregg.com/)
+- [Brendan D. Gregg: Linux bcc/BPF Node.js USDT Tracing](http://www.brendangregg.com/blog/2016-10-12/linux-bcc-nodejs-usdt.html)
+
+
+
+### Examples
+
+- The PR includes examples and documentation on how to run them.
+- For building Bitcoin Core with USDT support, you need the `sys/std.h` headers (when present, USDT support is [automatically] compiled in). On Debian-like systems you can install them with `apt install systemtap-sdt-dev` (this is not documented yet).
```

Comment:
I'm not sure what "you can install them" refers to here, installing headers seems an odd formulation. It auto-adds them?

`apt show systemtap-sdt-dev` returns this:

This package contains header file and executable (dtrace) that
 can be used to add static probes into userspace applications
 for use with SystemTap.

Reply:
I'm changing this to

```suggestion
- For building Bitcoin Core with USDT support, you need the `sys/std.h` headers (when present, USDT support is [automatically] compiled in). On Debian-like systems you can install the package `systemtap-sdt-dev` (this is not documented yet).
```